### PR TITLE
add Double & Dueling DQN

### PR DESCRIPTION
**Pull request by banma12956**

### CODE REVIEW

File: examples/DQN/train.py [hunk 1]
```diff
@@ -149,5 +165,6 @@ def main():
         help='the step interval between two consecutive evaluations')
 
     args = parser.parse_args()
-
+    rom_name = args.rom.split('/')[-1].split('.')[0]
```
Comment (TomorrowIsAnOtherDay) [inline examples/DQN/train.py]: We don't have to set the log directory explicitly.

File: parl/algorithms/fluid/ddqn.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        self.act_dim = act_dim
+        self.gamma = gamma
+
+    def predict(self, obs):
+        return self.model.value(obs)
+
+    def learn(self, obs, action, reward, next_obs, terminal, learning_rate):
+        pred_value = self.model.value(obs)
+        # layers.Print(action)
```
Comment (TomorrowIsAnOtherDay) [inline parl/algorithms/fluid/ddqn.py]: remove lines for debugging.

File: examples/DQN/atari_agent.py [hunk 3]
```diff
@@ -89,15 +95,18 @@ def learn(self, obs, act, reward, next_obs, terminal):
...
             'reward': reward,
             'next_obs': next_obs.astype('float32'),
-            'terminal': terminal
+            'terminal': terminal,
+            'lr': lr
         }
         cost = self.fluid_executor.run(
             self.learn_program, feed=feed, fetch_list=[self.cost])[0]
-        return cost
+        return cost, lr
```
Comment (TomorrowIsAnOtherDay) [inline examples/DQN/atari_agent.py]: Please do not return learning rate here.

File: examples/DQN/train.py [hunk 4]
```diff
@@ -137,6 +152,7 @@ def main():
         '--rom', help='path of the rom of the atari game', required=True)
     parser.add_argument(
         '--batch_size', type=int, default=64, help='batch size for training')
+    parser.add_argument('--algo', default='DQN', help='DQN/Double/Dueling DQN')
```
Comment (TomorrowIsAnOtherDay) [inline examples/DQN/train.py]: help='DQN/DDQN/Dueling, represent DQN, double DQN, and dueling DQN respectively'

File: examples/DQN/train.py [hunk 5]
```diff
@@ -137,6 +152,7 @@ def main():
...
     parser.add_argument(
         '--train_total_steps',
         type=int,
```
Comment (TomorrowIsAnOtherDay) [inline examples/DQN/train.py:160]: modify 1e8 as 1e7

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
Looks good to me.
